commit a9eab62d41646cbf5d8b3408c9d4617bb9678e71
Author: Lorenzo Bianconi <lorenzo.bianconi@redhat.com>
Date:   Fri Mar 9 10:41:41 2018 +0100

    mt7601u: let mac80211 validate rx CCMP PN
    
    Apparently the hardware does not perform CCMP PN validation so
    let mac80211 take care of possible replay attacks in sw.
    Moreover indicate ICV and MIC had been stripped setting corresponding
    bits in ieee80211_rx_status.
    The fix has been validated using 4.2.1 and 4.1.3 tests from the WiFi
    Alliance vulnerability detection tool.
    
    Fixes: c869f77d6abb ("add mt7601u driver")
    Acked-by: Jakub Kicinski <kubakici@wp.pl>
    Tested-by: David Park <david.park@hitemengineering.com>
    Signed-off-by: Lorenzo Bianconi <lorenzo.bianconi@redhat.com>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>

diff --git a/drivers/net/wireless/mediatek/mt7601u/initvals.h b/drivers/net/wireless/mediatek/mt7601u/initvals.h
index ec11ff66969d..2dc6b68e7fb9 100644
--- a/drivers/net/wireless/mediatek/mt7601u/initvals.h
+++ b/drivers/net/wireless/mediatek/mt7601u/initvals.h
@@ -139,6 +139,7 @@ static const struct mt76_reg_pair mac_common_vals[] = {
 	{ MT_TXOP_HLDR_ET,		0x00000002 },
 	{ MT_XIFS_TIME_CFG,		0x33a41010 },
 	{ MT_PWR_PIN_CFG,		0x00000000 },
+	{ MT_PN_PAD_MODE,		0x00000001 },
 };
 
 static const struct mt76_reg_pair mac_chip_vals[] = {